transformers/storage/median/repository.go

```diff
@@ -0,0 +1,286 @@
+	}
+}
+func (repository *MedianStorageRepository) SetDB(db *postgres.DB) {
+	repository.db = db
+}
+
+func (repository MedianStorageRepository) insertMedianVal(diffID, headerID int64, contractAddress, val string) error {
+	tx, txErr := repository.db.Beginx()
+	if txErr != nil {
+		return txErr
+	}
+
+	addressID, addressErr := shared.GetOrCreateAddress(contractAddress, repository.db)
+	if addressErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("contract address", addressErr)
+		}
+		return addressErr
+	}
+	_, insertErr := tx.Exec(insertMedianValQuery, diffID, headerID, addressID, val)
+	if insertErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("val record with address", insertErr)
+		}
+		return insertErr
+	}
+	return tx.Commit()
+}
+
+func (repository MedianStorageRepository) insertMedianAge(diffID, headerID int64, contractAddress, age string) error {
+	tx, txErr := repository.db.Beginx()
+	if txErr != nil {
+		return txErr
+	}
+
+	addressID, addressErr := shared.GetOrCreateAddress(contractAddress, repository.db)
+	if addressErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("contract address", addressErr)
+		}
+		return addressErr
+	}
+	_, insertErr := tx.Exec(insertMedianAgeQuery, diffID, headerID, addressID, age)
+	if insertErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("age record with address", insertErr)
+		}
+		return insertErr
+	}
+	return tx.Commit()
+}
+
+func (repository MedianStorageRepository) insertMedianBar(diffID, headerID int64, contractAddress, bar string) error {
+	tx, txErr := repository.db.Beginx()
+	if txErr != nil {
+		return txErr
+	}
+
+	addressID, addressErr := shared.GetOrCreateAddress(contractAddress, repository.db)
+	if addressErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("contract address", addressErr)
+		}
+		return addressErr
+	}
+	_, insertErr := tx.Exec(insertMedianBarQuery, diffID, headerID, addressID, bar)
+	if insertErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("bar record with address", insertErr)
+		}
+		return insertErr
+	}
+	return tx.Commit()
+}
+
+func (repository *MedianStorageRepository) insertPackedValueRecord(diffID, headerID int64, contractAddress string, metadata types.ValueMetadata, packedValues map[int]string) error {
+	for order, value := range packedValues {
+		var insertErr error
+		switch metadata.PackedNames[order] {
+		case Val:
+			insertErr = repository.insertMedianVal(diffID, headerID, contractAddress, value)
+		case Age:
+			insertErr = repository.insertMedianAge(diffID, headerID, contractAddress, value)
+		default:
+			panic(fmt.Sprintf("unrecognized median contract storage name in packed values: %s", metadata.Name))
```

Comment:
I know we've been using `panic`s in a few places like this, but maybe we could return the error here?